app/common/model/src/main/java/io/syndesis/common/model/integration/step/template/AbstractTemplatePreProcessor.java

```diff
@@ -231,7 +231,11 @@ public void reset() {
 
     @Override
     public Map<String, Object> getUriParams() {
-        return Collections.emptyMap();
+        // Since Camel 3.3 we must declare explicitly the usage of templates
+        // that are stored in headers
+        Map<String, Object> defaultMap = new HashMap<>();
+        defaultMap.put("allowTemplateFromHeader","true");
+        return defaultMap;
```

Comment:
This may be replaced by `Collections.singletonMap("allowTemplateFromHeader","true")` ?

Reply:
It was my first choice, but that would return an immutable map and the returned map will be updated by the caller.

Reply:
wonder if those parameter should be set at construction time since it seems they are constants

Reply:
I see. What if we have a static singleton map and we return in that method a new collection filled with the singleton?

Reply:
up to you